extmod/modussl_mbedtls.c

```diff
@@ -148,7 +167,18 @@ STATIC mp_obj_ssl_socket_t *socket_new(mp_obj_t sock, struct ssl_args *args) {
         goto cleanup;
     }
 
-    mbedtls_ssl_conf_authmode(&o->conf, MBEDTLS_SSL_VERIFY_NONE);
+    if (psk_mode) {
+        const char *ident = mp_obj_str_get_str(args->psk_ident.u_obj);
+        size_t key_len;
+        const byte *key = (const byte*)mp_obj_str_get_data(args->psk_key.u_obj, &key_len);
+        //if (key_len > MBEDTLS_PSK_MAX_LEN) {
```

Comment:
Should probably keep this check and raise a `ValueError`, also for the ident, but it should be in the beginning of this function.

Reply:
I don't remember why I left that in there. My intention was that this would be caught by mbedtls and hit line 251 `ret == MBEDTLS_ERR_PK_BAD_INPUT_DATA`.

Reply:
mbedtls errors can be a bit obscure. Especially when it's directly related to user input, I think a concise exception is preferred.

Reply:
agreed, but did you look at the code?
```
    } else if (ret == MBEDTLS_ERR_PK_BAD_INPUT_DATA) {
        mp_raise_ValueError("invalid key");
```
this catches any issue with the key, not just "too long". It's a bit a balance between the number or different error messages and the space they end up using.

Reply:
Haha I admit, looked only on your diff. That's fine. Just need to remove that comment.